### PR TITLE
Remove confusing placeholder in Sharing Settings

### PR DESCRIPTION
**Pull request by matthiasbe**
Hi,
I would like to provide a solution for issue  #19806 

I am new on this project, as a quick introduction I just installed the server at home and found it wounderful. As I know a bit of web development I would like to help improving it.

Best
Matthias

### CODE REVIEW

**Comment by gary-kim** [comment]
@jancborchardt I think you said you'd rather have 7 be a default but do you think this is okay?

**Comment by jancborchardt** [comment]
> @jancborchardt I think you said you'd rather have 7 be a default but do you think this is okay?

Yeah, I would still say it’s good if there is a default suggestion, and for that to be 7. :)

Otherwise everyone who wants to use this would have to check the checkmark and then put in a number. With a default, if you think 7 is reasonable (which we decided it often is) then it’s just one enabling click.

**Comment by matthiasbe** [comment]
If I understand correctly, you want the following

When I first click on "set default expiration date for link shares" I get this
image
Days amount is prefilled with "7" but "Enforce expiration date" is unticked

If I got this right, we run into the problem stated in #19806, analoguous to the current version with the placeholder. It looks like the default value is 7 even though the "7" is just a suggestion.

**Comment by jancborchardt** [comment]
> If I got this right, we run into the problem stated in #19806, analoguous to the current version with the placeholder. It looks like the default value is 7 even though the "7" is just a suggestion.

The problem in that issue is: `The "7" in the field is slightly greyed out and effectively NOT set.` – by simply actually using it as the value, we fix the issue. That’s the only thing we need to fix. :)

**Comment by matthiasbe** [comment]
@jancborchardt I pushed a new commit according to your indications.

If no value is set and "Set default expiration date for link shares" is ticked, values is set to 7 and updated on the server

I used jQuery's `trigger("change")` to trigger the updating event.

**Comment by jancborchardt** [comment]
> If no value is set and "Set default expiration date for link shares" is ticked, values is set to 7 and updated on the server

What happens if you remove the value? It’s just set to 7 again? It’s a bit strange of an interaction – it would be better to just leave the placeholder of "7" there, and if it’s empty use 7 as value. But don’t fill it in the field.

**Comment by matthiasbe** [comment]
I see your point. What about whenever value is emptied by user, instead of a placeholder '7' is placed back in the field and sent to the server ? Or do you really prefer the placeholder ?

**Comment by jancborchardt** [comment]
Yup, a placeholder is preferred, as it’s weird behavior to fill a field with a value when people didn’t put it in. :) A placeholder will communicate it much better.

**Comment by matthiasbe** [comment]
Ok thank you for the extra precision.
Here is the full behavior:

* If "Set default expiration date for link shares" is ticked, the value is sent to the server.
* Whenever the value is empty, `7` is sent to the server

Note that if the box is ticked without any value, and then page is reloaded, the value will be `7` (not the placeholder) because the value is filled on server-side.

**Review by jancborchardt — APPROVED** [review]
Seems good design-wise now. :) cc @skjnldsv for server

File: apps/settings/js/admin.js [hunk 1]
```diff
@@ -79,15 +79,24 @@ $(document).ready(function(){
 				value = 'no';
 			}
 		}
+		if ((this.id === 'shareapiExpireAfterNDays' || this.id === 'shareapiInternalExpireAfterNDays') && value === '') {
+			value = '7'
```
Comment (skjnldsv) [inline apps/settings/js/admin.js:83]: I guess we have an available value for this somewhere? Hardcoded numbers like those should be avoided

Comment (matthiasbe) [reply apps/settings/js/admin.js:83]: I agree, however I couldn't find any constant for this. Do you have any pointer on where to add this ?

Comment (skjnldsv) [reply apps/settings/js/admin.js:83]: Sorry, I missed this comment.
Maybe in the capabilities?

Comment (szaimen) [reply apps/settings/js/admin.js:83]: @matthiasbe any update here? :)

**Comment by skjnldsv** [comment]
As there is no feedback since a while I will close this PR. If you are still willing to get this in, please address the potential comments and rebase to latest master. Then, feel free to re-open.

**Comment by matthiasbe** [comment]
Hello, sorry I didn't give any update. After a while it's hard to get back into the code. Closing is the best thing to do for now, thanks.

**Pull request closed without merging**